File: solutions/122.Best_Time_to_Buy_and_Sell_Stock_II/AC_greedy_n.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_greedy_n.cpp
*  Create Date: 2015-03-12 11:10:07
*  Descripton:   
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int profit = 0;
        for (int i = 1; i < prices.size(); ++i)
            if (prices[i] > prices[i - 1])
                profit += prices[i] - prices[i - 1];
        return profit;
    }
};

int main() {

	return 0;
}

File: solutions/122.Best_Time_to_Buy_and_Sell_Stock_II/AC_greedy_n.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_greedy_n.py
# Create Date: 2015-03-12 11:14:05
# Usage:       AC_greedy_n.py 
# Descripton:  


class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        return sum(y - x for x, y in zip(prices[:-1], prices[1:]) if x < y)

# debug
s = Solution()
print s.maxProfit([1, 3, 4])

